src/concepts/essentials/binary.md

```diff
@@ -0,0 +1,74 @@
+# Dealing with Binary Data
+
+Caido plugins are written in JavaScript. When creating a plugin that must handle raw bytes, there are some caveats to be aware of as the backend language used by Caido is Rust.
+
+## UTF-8 Encoding
+
+For Unicode code points outside of the ASCII range <span style="color: #EBEBF599; font-style: italic">(U+0000 to U+007F)</span> multiple bytes are used.
+
+When encoding code points into multibyte UTF-8 characters, the bytes are prefixed with specific bit patterns. The first byte prefix indicates how many bytes are used, and any subsequent bytes begin with 10.
+
+<img alt="UTF-8 encoding chart." src="/_images/utf_chart.png" center/>
+
+This can be an issue if you want to create a Caido plugin that sends binary to a target server to see how it is processed.
+
+## Example
+
+Consider this `/packages/backend/src/index.ts` file:
+
+``` ts
+import { RequestSpecRaw } from "caido:utils";
+import { SDK, DefineAPI } from "caido:plugin";
+
+async function testSendRequest(sdk: SDK): Promise<void> {
+  console.log("Testing send request");
+  const req = new RequestSpecRaw("http://localhost:5555");
+  req.setRaw("GET /admin\x85 HTTP/1.1\r\nHost: HOST\r\n\r\n");
+
+  const res = await sdk.requests.send(req);
+  console.log(res?.response.getRaw().toText());
+}
+```
+
+The intention is to send the byte `\x85` with a binary value of `[1000 0101]`. However, JavaScript is interpreting it as the Unicode code point.
+
+::: info This happens because:
+
+1. JavaScript strings are UTF-16 encoded.
+2. `\x85` in a string becomes `U+0085` in UTF-16 (`[0000 0000 1000 0101]`).
+3. When passed to Rust, `U+0085` is encoded as UTF-8:
+   - First byte `C2`: `[11000010]` <span style="color: #EBEBF599; font-style: italic">(prefix marks start of 2-byte sequence)</span>
+   - Second byte `85`: `[10000101]` <span style="color: #EBEBF599; font-style: italic">(continuation byte)</span>
+:::
+
+This will result in it being interpreted as the unprintable `NEL` <span style="color: #EBEBF599; font-style: italic">(Next Line)</span> character, which will be replaced with `�`.
```

Comment:
This sentence plus the tip make no sense in context, it was probably moved around or something before it was removed.
You just said that this will send C2 85 and NOT just 85. This exemple is when the actual character 85 is sent.

Reply:
If you want to show what happens if you dont deal with bytes correctly, then you would should xxd capture with C2 85